src/lightning/pytorch/strategies/xla.py

```diff
@@ -265,18 +265,21 @@ def reduce(
 
         return output
 
+    @override
+    def setup_environment(self) -> None:
+        self._launched = True
+        super().setup_environment()
+
     @override
     def setup_distributed(self) -> None:
         assert self.parallel_devices is not None
-        if _using_pjrt() and len(self.parallel_devices) == 1:
+        if len(self.parallel_devices) == 1:
             # spawning only 1 device with PjRT is not supported:
             # https://github.com/Lightning-AI/lightning/pull/17408#discussion_r1170671732
             raise NotImplementedError(
                 "The `XLAStrategy` does not support running on a single device with the PjRT runtime."
                 " Try using all devices or the `SingleDeviceXLAStrategy` strategy"
             )
-
-        self._launched = True
```

Comment:
I moved this up to setup_environment(). The #19184 causes root_device to be called too early so that `self._launched` is not yet set, which was the reason why the tests in CI failed. 

We can break this into a separate PR if you prefer.